frontend/src/component/providers/AccessProvider/AccessProvider.tsx

```diff
@@ -70,22 +70,12 @@ const checkPermission = (
     if (
         p.permission === permission &&
         (p.project === project || p.project === '*') &&
-        (p.environment === environment || p.environment === '*')
+        (!p.environment ||
+            p.environment === environment ||
+            p.environment === '*')
     ) {
         return true;
     }
 
-    if (
-        p.permission === permission &&
-        (p.project === project || p.project === '*') &&
-        !Boolean(p.environment)
-    ) {
-        return true;
-    }
-
-    return (
-        p.permission === permission &&
-        p.project === undefined &&
-        p.environment === null
-    );
+    return p.permission === permission && !p.project && !p.environment;
```

Comment:
My use-case was to check both "root" and "project" permissions at the same time (e.g. `permission={[UPDATE_SEGMENT, UPDATE_PROJECT_SEGMENT]}`), however my "root" permission was not correctly evaluated when specifying a `project`. E.g. see: https://github.com/Unleash/unleash/pull/3346/files#diff-09482643904b9601b01cf445d86cbc5fc1c589ed115ce166891ae67358a940a6R23-R24

This change seems to make that possible without breaking anything (hopefully). From my understanding, `p.environment` may be returned as `undefined` now instead of `null`, so we simply check if those fields are falsy instead, which should mean that the underlying permission doesn't care about which project or environment we sent here to check, which should mean we're evaluating a root permission.

Reply:
Note this also follows the same logic as the backend: https://github.com/Unleash/unleash/blob/6b165f36635fd633e9c92c163e14593249cd8437/src/lib/services/access-service.ts#L120-L123